store/src/java/com/zimbra/cs/event/logger/EventLogger.java

```diff
@@ -197,29 +210,32 @@ private EventNotifier(Map<String, EventLogHandler.Factory> knownFactories, Map<S
         @Override
         public void run() {
             try {
-                while (!Thread.currentThread().isInterrupted()) {
-                    consume(eventQueue);
+                Event event = null;
+                while (!shutdownExecutor.get()) { //don't process any more events if shutdownExecutor is set to true.
+                    /* poll is used so we can check if shutdownExecutor is set every 3 seconds.
+                    if take was used the code would block on it in case of empty event queue. */
+                    event = eventQueue.poll(POLL_TIMEOUT, TimeUnit.SECONDS);
```

Comment:
Is there any way to do something like this in a way where it gets stopped when we know that we want to shutdown?  Un-necessary 3 second delays considered harmful.

Reply:
@fatkudu I could not think of a way to do it without using a poll method.